test/images/agnhost/connect/connect.go

```diff
@@ -114,11 +120,24 @@ func connectSCTP(dest string, timeout time.Duration) {
 	errCh := make(chan error)
 
 	go func() {
-		conn, err := sctp.DialSCTP("sctp", nil, addr)
-		if err == nil {
-			conn.Close()
+		conn, err = sctp.DialSCTP("sctp", nil, addr)
+		if err != nil {
+			errCh <- err
+			return
+		}
+		defer func() {
+			errCh <- conn.Close()
+		}()
```

Comment:
This implementation with the goroutine is hard to read, since is test code I Assume is ok, but if you return because of any error this will try to send something for a channel that nothing is reading, however the previous error will call os.Exit so this will die independently